nifi-nar-bundles/nifi-cdc/nifi-cdc-postgresql-bundle/nifi-cdc-postgresql-processors/src/test/java/org/apache/nifi/cdc/postgresql/processors/ITCaptureChangePostgreSQL.java

```diff
@@ -0,0 +1,1113 @@
+
+/**
+ * Integration Test for PostgreSQL CDC Processor.
+ *
+ * The PostgreSQL Cluster should be configured to enable logical replication:
+ *
+ * - Property listen_addresses should be set to the hostname to listen to;
+ *
+ * - Property max_wal_senders should be at least equal to the number of
+ * replication consumers;
+ *
+ * - Property wal_keep_size (or wal_keep_segments for PostgreSQL versions
+ * before 13) specifies the minimum size of past WAL segments kept in the WAL
+ * dir;
+ *
+ * - Property wal_level for logical replication should be equal to
+ * logical;
+ *
+ * - Property max_replication_slots should be greater than zero for logical
+ * replication.
+ *
+ * For example:
+ *
+ * <code>
+ * [postgresql.conf]
+ *
+ * listen_addresses = '*'
+ * max_wal_senders = 4
+ * wal_keep_size = 4
+ * wal_level = logical
+ * max_replication_slots = 4
+ * </code>
+ *
+ * For the Integration Tests it will be necessary to create specific users and
+ * database:
+ *
+ * <code>
+ * $ psql -h localhost -p 5432 -U postgres
+ *
+ * CREATE USER nifi SUPERUSER PASSWORD 'Change1t!';
+ * CREATE USER intern PASSWORD 'justForT3sts!';
+ *
+ * CREATE DATABASE db_test;
+ * </code>
+ *
+ * At last, authorize query and replication connections for user(s) and host(s).
+ * For example:
+ *
+ * <code>
+ * [pg_hba.conf]
+ *
+ * host db_test nifi 192.168.56.0/24 md5
+ * host db_test intern 192.168.56.0/24 md5
+ *
+ * host replication nifi 192.168.56.0/24 md5
+ * host replication intern 192.168.56.0/24 md5
+ * </code>
+ *
+ * These configurations requires the restart of PostgreSQL service.
+ */
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class ITCaptureChangePostgreSQL {
+
+        // Change it!
+        private static final String DRIVER_LOCATION = "/Users/davy.machado/Documents/SOFTWARES/LIBS/postgresql-42.3.1.jar";
+        private static final String DRIVER_NAME = "org.postgresql.Driver";
+        private static final String HOST = "192.168.56.101";
+        private static final String PORT = "5432";
+        private static final String DATABASE = "db_test";
+        private static final String USERNAME = "nifi";
+        private static final String PASSWORD = "Change1t!";
+        private static final String CONNECTION_TIMEOUT = "30 seconds";
+        private static final String PUBLICATION = "pub_city";
+        private static final String REPLICATION_SLOT = "slot_city";
+
+        ITPostgreSQLClient client;
+        TestRunner runner;
+
+        @Before
+        public void setUp() throws Exception {
+                runner = TestRunners.newTestRunner(new CaptureChangePostgreSQL());
+                client = new ITPostgreSQLClient(HOST, PORT, DATABASE, USERNAME, PASSWORD);
+        }
+
+        @After
+        public void tearDown() throws Exception {
+                runner = null;
+
+                if (client != null) {
+                        client.closeConnection();
+                        client = null;
+                }
+        }
```

Comment:
The indentation formatting for this file also appears incorrect and needs to be changed to four spaces.

Reply:
The same before. I'll update the indentation to four spaces.

Reply:
Done!